scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: $t_metric_programmable_param_value_exp_t*
+      name: pParameterValues
+      desc: "[in] list of parameter values to be set."
+    - type: uint32_t
+      name: parameterCount
+      desc: "[in] Count of parameters to set."
+    - type: const char
+      name: "name[ZET_MAX_METRIC_NAME]"
+      desc: "[in] Metric name to be used."
+    - type: const char
+      name: "description[ZET_MAX_METRIC_DESCRIPTION]"
+      desc: "[in] Metric description to be used."
+    - type: "uint32_t*"
+      name: pMetricHandleCount
+      desc: |
+            [in,out] Pointer to the number of metric handles.
+            if count is zero, then the driver shall update the value with the number of metric handles available for this programmable.
+            if count is greater than the number of metric handles available, then the driver shall update the value with the correct number of metric handles available.
+    - type: "$t_metric_handle_t*"
+      name: phMetricHandles
+      desc: |
+            [in,out][optional][range(0,*pMetricHandleCount)] array of handle of metrics.
+            if count is less than the number of metrics available, then driver shall only retrieve that number of metric handles.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create metric group handle."
+version: "1.8"
+class: $tMetricGroup
+name: CreateExp
+decl: static
+details:
+    - "Metric handles from multiple domains may be used for a single metric group."
+    - "Metric handles from different sourceIds (refer $t_metric_programmable_exp_properties_t) are not allowed in a single metric group."
+params:
+    - type: $t_device_handle_t
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: const char
+      name: name[ZET_MAX_METRIC_GROUP_NAME]
+      desc: "[in] Metric group name."
+    - type: const char
+      name: description[ZET_MAX_METRIC_GROUP_DESCRIPTION]
+      desc: "[in] Metric group description."
+    - type: $t_metric_group_handle_t*
+      name: phMetricGroup
+      desc: "[out] Created Metric group handle"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Add a metric handle to the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: AddMetricExp
+decl: static
+details:
+    - "Reasons for failing to add the metric could be queried using pErrorString"
+params:
+    - type: $t_metric_group_handle_t 
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric to be added to the group."
+    - type: "uint32_t*"
+      name: pErrorStringSize
+      desc: "[in,out][optional] Size of the error string to query, if an error was reported during adding the metric handle."
```

Comment:
missing something like "mist be 0 if pErrorString == nullptr"  like in 
https://spec.oneapi.io/level-zero/latest/tools/api.html#zetcontextactivatemetricgroups

Reply:
updated the explanation for size==0 case